### PR TITLE
Mac PW Pool: Install libkrun (krunkit)

### PR DESCRIPTION
**Pull request by cevich**
In order to test accessibility of the host GPU inside a podman machine
container, it's necessary to install support for krun.  However, since
the list of brew recipes is ever growing, split it up into sections with
comments explaining why each is necessary and what uses it.

Ref: [Test PR](https://github.com/containers/automation/pull/208)

### CODE REVIEW

**Comment by cevich** [comment]
Observed in `setup.log`:

```
...cut...
==> Tapping cfergeau/crc
Cloning into '/opt/homebrew/Library/Taps/cfergeau/homebrew-crc'...
Tapped 3 formulae (16 files, 82.7KB).
==> Fetching cfergeau/crc/vfkit
==> Cloning https://github.com/crc-org/vfkit.git
Cloning into '/Users/ec2-user/Library/Caches/Homebrew/vfkit--git'...
==> Checking out tag v0.5.1
HEAD is now at 441c134 vfkit 0.5.1 release
==> Installing vfkit from cfergeau/crc
...cut...
```

**Comment by cevich** [comment]
If I manually run `brew install slp/krunkit/krunkit` I get:

```
==> Tapping slp/krunkit
Cloning into '/opt/homebrew/Library/Taps/slp/homebrew-krunkit'...
remote: Enumerating objects: 53, done.
remote: Counting objects: 100% (53/53), done.
remote: Compressing objects: 100% (40/40), done.
remote: Total 53 (delta 18), reused 48 (delta 13), pack-reused 0
Receiving objects: 100% (53/53), 5.01 MiB | 32.49 MiB/s, done.
Resolving deltas: 100% (18/18), done.
Tapped 3 formulae (20 files, 10MB).
==> Fetching dependencies for slp/krunkit/krunkit: dtc, libepoxy, molten-vk, slp/krunkit/virglrenderer and slp/krunkit/libkrun-efi
...cut...
==> Installing slp/krunkit/krunkit
==> Pouring krunkit-0.1.1.arm64_sonoma.bottle.tar.gz
Error: The `brew link` step did not complete successfully
The formula built, but is not symlinked into /opt/homebrew
Could not symlink bin/vfkit
Target /opt/homebrew/bin/vfkit
is a symlink belonging to vfkit. You can unlink it:
  brew unlink vfkit

To force the link and overwrite all conflicting files:
  brew link --overwrite krunkit

To list all files that would be deleted:
  brew link --overwrite krunkit --dry-run

Possible conflicting files are:
/opt/homebrew/bin/vfkit -> /opt/homebrew/Cellar/vfkit/0.5.1/bin/vfkit
==> Summary
🍺  /opt/homebrew/Cellar/krunkit/0.1.1: 7 files, 1.2MB
==> Running `brew cleanup krunkit`...
Disable this behaviour by setting HOMEBREW_NO_INSTALL_CLEANUP.
Hide these hints with HOMEBREW_NO_ENV_HINTS (see `man brew`).
ip-172-31-8-253:~ ec2-user$ echo $?
1
```

**Comment by cevich** [comment]
FWIW, the fix was `brew tap slp/krun` -> `brew tap slp/krunkit`

**Comment by cevich** [comment]
This is ready to be reviewed/merge but I'm leaving this as "draft" until I return from PTO.  i.e. I don't want these changes to go into the worker-pool and possible mess something up while I'm away.

@edsantiago and/or @ashley-cui if you want to give this an initial look over, it's not required but always appreciated :smile:

File: mac_pw_pool/setup.sh
Comment (cevich) [inline mac_pw_pool/setup.sh:93]: Note to me: Do a test-run of the setup again to see if this has been fixed from a `brew install` perspective.

File: mac_pw_pool/setup.sh
Comment (edsantiago) [inline mac_pw_pool/setup.sh:178]: Taking a step back, is it really worth preserving the `if`? You're doing `ln -sf` (force) below, so even if this runs twice it should be ok. And it would really simplify life for a maintainer: no need to think about the conditional.

**Review by edsantiago — APPROVED** [review]
LGTM assuming it does all the MacNecessary things. Two questions inline, neither blockers. Thanks for making this a simple clean one.

File: mac_pw_pool/setup.sh
Comment (cevich) [reply mac_pw_pool/setup.sh:77]: Bump was based on observation w/ addition of 'krunkit' and breaking up the install command.  Brew produces quite a bit of output during each install, and install order is significant.  Like DNF repos., download times can also vary quite a bit due to many factors.

I chose to break up the installs in this PR mainly due to the [slp/homebrew-krunkit](https://github.com/slp/homebrew-krunkit/pull/1) problem I ran into.  This appears to have been resolved now, so I could switch to a package array w/ line comments and a single install command.

I'll give it one try, but if it fails this will need to be done as a followup.  The testing-feature is needed somewhat urgently and I won't be able to monkey around with the 3-hour DH spinup/spindown delays.

File: mac_pw_pool/setup.sh
Comment (cevich) [reply mac_pw_pool/setup.sh:178]: Good point.  I can change this easy enough.  It's such a toungue-in-cheek "fix" anyway, best keep it as simple as possible.

**Comment by edsantiago** [comment]
LGTM since this is urgent and my questions are all minor.

File: mac_pw_pool/setup.sh
Comment (cevich) [reply mac_pw_pool/setup.sh:178]: Investigated doing this, it's not worth it.  The setup script needs to be able to run more than once on a mac.  This symlink's existence "protecting" execution of a few apple-specific config. commands where retrieving their current state is complex.  I could use a separate marker-file, but didn't think there was much point.

I'll add a comment detailing this for future maintainers.

**Review by ashley-cui — COMMENTED** [review]
LGTM, I'm okay with the brew cleanup in a followup PR  :)

**Comment by cevich** [comment]
Thanks @edsantiago and @ashley-cui.

I'll give it a college-try at making the tap/formula part simpler.  If it fails I'll revert.

**Comment by cevich** [comment]
force-push: Comment updates + attempt at simplifying tap/formula install.

**Comment by cevich** [comment]
FWIW, this is the setup.log from running the script in 783ae1e

[setup.log](https://github.com/user-attachments/files/16446046/setup.log)

File: mac_pw_pool/setup.sh
Comment (cevich) [reply mac_pw_pool/setup.sh:81]: FWIW, this is the setup.log from running the script in 783ae1e

[setup.log](https://github.com/user-attachments/files/16446046/setup.log)

**Comment by cevich** [comment]
This is [passing all tests](https://github.com/containers/podman/pull/23290#issuecomment-2261345177).  I'm going to rebase then assuming CI (here) is happy, merge so the pool can start using the new setup.

**Comment by cevich** [comment]
Manually updated maintenance VM, so all newly created Macs will use these updates.

**Pull request merged**
